### PR TITLE
[WFLY-18526] adds proposal to move BOMs to the Server source repository

### PR DESCRIPTION
**Pull request by emmartins**
Issue: https://issues.redhat.com/browse/WFLY-18526

### CODE REVIEW

File: boms/WFLY-18526_Move_BOMs_to_the_Server_Repository.adoc [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+// to that kind of WildFly installation
+
+* [] Bootable jar (local and cloud deployment)
+* [] OpenShift S2I
+
+== Requirements
+
+=== Hard Requirements
+
+* The BOMs sources should be moved to the Server repository, and placed in a new /boms/user directory. The BOMs sources repository history should be included in such move.
```
Comment (jamezp) [inline boms/WFLY-18526_Move_BOMs_to_the_Server_Repository.adoc:65]: "Server" is used a lot here. By "server repository" do you mean the WildFly repository?

**Review by jamezp — COMMENTED** [review]
I don't recall when this was done, but we originally pulled the BOM out of the WildFly repo. TBH I don't recall the reason either :) I think it was a chicken and egg type of issue for the Quickstarts.

That said, I was never personally a fan of using the `org.wildfly:wildfly-parent` POM to import the dependencies. I do understand why we did that though.

I'm personally not opposed to moving the BOM back into the main WildFly repository. However, I think we need to take care when we do it. I don't think we really need to preserve the history, but that's just my opinion.

File: boms/WFLY-18526_Move_BOMs_to_the_Server_Repository.adoc (hunk 1 above)
Comment (emmartins) [reply boms/WFLY-18526_Move_BOMs_to_the_Server_Repository.adoc:65]: yes, "server" is used an abstraction for WildFly and EAP

**Comment by emmartins** [comment]
> I don't recall when this was done, but we originally pulled the BOM out of the WildFly repo. TBH I don't recall the reason either :) I think it was a chicken and egg type of issue for the Quickstarts.
> 

IIRC it was more of a technical issue but we had a very different solution at that time, with static built BOMs.

> That said, I was never personally a fan of using the `org.wildfly:wildfly-parent` POM to import the dependencies. I do understand why we did that though.

The WildFly BOMs now uses the - fairly recent - internal BOMs to obtain the dep management but it is not much diference TBH, but what we are doing is totally different than in the past, we now have BOM Builders instead, which use the maven plugin taking advantage of the whole dep management "context", to restore the transitive relationships lost with the "excludes".

> I'm personally not opposed to moving the BOM back into the main WildFly repository. However, I think we need to take care when we do it. I don't think we really need to preserve the history, but that's just my opinion.

FYI the feature PR is merged already.

**Comment by jamezp** [comment]
Oh, should this be merged then?

**Comment by emmartins** [comment]
Let's keep it open till feature team approves it, if changes or enhancements are needed we can add another feature PR. Downstream this only targets 8.1 so let's give time everyone some time to digest it.

**Comment by bstansberry** [comment]
Time enough to digest. :) Time to merge.

**Pull request merged**